pkg/kubelet/status/status_manager.go

```diff
@@ -232,8 +240,33 @@ func (m *manager) State() state.Reader {
 	return m.state
 }
 
+// GetContainerResourceAllocation returns the last checkpointed ResourcesAllocated values
+// If checkpoint manager has not been initialized, it returns nil, false
+func (m *manager) GetContainerResourceAllocation(podUID string, containerName string) (v1.ResourceList, bool) {
```

Comment:
check nil and return not found before locking?

Reply:
yes.

Reply:
state will still be nil in all of these functions if the feature gate is disabled. 

sweeping all the callers, it's pretty twisty to tell that all of the call sites are guarded by feature gate checks (these are sometimes called from functions which are only called under feature gate guard, or are called from declared closures which are then invoked inside a feature gate guard)

Reply:
this PR is an improvement on current state of master, but this still seems like a sharp edge to improve before 1.27 cut

Reply:
True. I can add feature gates check in these functions as a conservative check that makes it easier for future reviews vs. adding a nil check and forgetting to remove at GA. It is one extra compare and branch instruction we don't need on every pod sync (if we can count on pull & CI tests).

Reply:
> this PR is an improvement on current state of master, but this still seems like a sharp edge to improve before 1.27 cut

What other things do you want to see fixed? I initially wanted to move away from node local checkpoint in favor of relying on values persisted in PodStatus (if it is legit use of [KEP 2527](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/2527-clarify-status-observations-vs-rbac)), but that wouldn't fly in standalone kubelet scenario. I feel kubelet could use a more generic KV abstraction that better decouples the checkpointing implementation details, but that is a bit larger scope work.. jmho.

Reply:
probably not leaving those manager methods NPE landmines if the feature-gate is disabled? either doing nil checks in those functions, or assigning a dummy/no-op impl to `state` when the feature gate is off

but a follow-up is fine... I'm more concerned with getting alpha standalone clusters green again at this point

Reply:
no-op is a good idea. Since this fix is urgent and already LTGM'd, I'll create a separate PR follow-up or tack it on to an existing follow-up PR.

Reply:
Does PR https://github.com/kubernetes/kubernetes/pull/116351 get it done?